Sources/Amplitude/AMPDeviceInfo.m

```diff
@@ -101,27 +101,48 @@ - (NSString *)model {
 }
 
 - (NSString *)carrier {
+    // unable to fetch carrier information
     if (!_carrier) {
-        Class CTTelephonyNetworkInfo = NSClassFromString(@"CTTelephonyNetworkInfo");
-        SEL subscriberCellularProvider = NSSelectorFromString(@"subscriberCellularProvider");
-        SEL carrierName = NSSelectorFromString(@"carrierName");
-        if (CTTelephonyNetworkInfo && subscriberCellularProvider && carrierName) {
-            networkInfo = [[NSClassFromString(@"CTTelephonyNetworkInfo") alloc] init];
-            id carrier = nil;
-            id (*imp1)(id, SEL) = (id (*)(id, SEL))[networkInfo methodForSelector:subscriberCellularProvider];
-            if (imp1) {
-                carrier = imp1(networkInfo, subscriberCellularProvider);
-            }
-            NSString *(*imp2)(id, SEL) = (NSString *(*)(id, SEL))[carrier methodForSelector:carrierName];
-            if (imp2) {
-                _carrier = imp2(carrier, carrierName);
+        _carrier = @"Unknown";
+    }
+
+    Class CTTelephonyNetworkInfo = NSClassFromString(@"CTTelephonyNetworkInfo");
+    if (!CTTelephonyNetworkInfo) {
+        return _carrier;
+    }
+
+    networkInfo = [[CTTelephonyNetworkInfo alloc] init];
```

Comment:
You should check if there is an instance already, otherwise, it will flood the console with warnings.